descqa/SizeStellarMassLuminosity.py

```diff
@@ -90,12 +98,26 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
         catalog_data = catalog_instance.get_quantities(list(colnames.values()), filters=filters)
         catalog_data = {k: catalog_data[v] for k, v in colnames.items()}
 
-        fig, axes = plt.subplots(2, 3, figsize=(9, 6), sharex=True, sharey=True)
+
+        fig, axes = plt.subplots(self.fig_subplot_row, self.fig_subplot_col, figsize=(self.fig_subplot_col*4, self.fig_subplot_row*4), sharex=True, sharey=True)
+
+        for n in range(len(self.catalogs)):
+            if catalog_name == self.catalogs[n]: # and self.observation == 'onecomp':
+                catalog = self.catalogs[n]
+                ylim = self.ylims[n] #[3e-1, 20]     
```

Comment:
Line 104-107: Do not use catalog name to set y-axis limits. The current implementation assumes catalogs with certain names will have certain range of size values. This is, however, not always true, and the test should be agnostic about the catalog. If you want to adjust the y-axis limits for different catalogs, you can do that by, for example, measuring the 5% and 95% quantiles of the size values and then set y-axis limits accordingly.